src/KGBotka/Calc.hs

```diff
@@ -0,0 +1,210 @@
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE LambdaCase #-}
+
+module KGBotka.Calc where
+
+import Control.Applicative (Alternative(..))
+import Control.Monad (void)
+import Control.Monad.Trans.Class (lift)
+import Control.Monad.Trans.Except (ExceptT(..), throwE)
+import Data.Char (isAlpha, isDigit)
+import qualified Data.Map.Strict as M
+import qualified Data.Text as T
+import KGBotka.Parser
+import System.Random (randomIO)
+
+data Operator
+  = Add
+  | Sub
+  | Mul
+  | Div
+  | Mod
+  | Pow
+  deriving (Show, Eq)
+
+charToOperator :: Char -> Operator
+charToOperator '+' = Add
+charToOperator '-' = Sub
+charToOperator '*' = Mul
+charToOperator '/' = Div
+charToOperator '%' = Mod
+charToOperator '^' = Pow
+charToOperator _ =
+    -- TODO: Make charToOperator not throw an exception
+    -- For now it is an unsafe function because it indicates
+    -- that operators were added but the parsers were not
+    -- corrected.
+  error
+    "charToOperator: No pattern for operator conversion. \
+    \Maybe you added a new operator and forgot to add a case to the conversion function."
```

Comment:
So, right now if I provide incorrect operator that will crash the entire thread this evaluation is running in?

Reply:
@rexim I did this because this crash indicates to whoever added an operator that this function was not corrected. In other words: the possibly added functionality implementation is incomplete.
See e.g. `parseAdditive`: it is impossible for the user in the chat to trigger this exception if nothing in this code has been changed. On the other hand, there should be a way to catch this kind of error at compile time, but I don't see it right now.

Reply:
Fixed as of a4ee58a